basic-2/src/test/java/PageObject.java

```diff
@@ -0,0 +1,51 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public abstract class PageObject {
+		protected ChromeDriver driver;
+		protected String baseUrl;
+		protected static long ATTRIBUTE_TO_MATCH_TIMEOUT_SECONDS= 5;
+		
+		protected PageObject(ChromeDriver driver, String baseUrl) {
+			this.driver = driver;
+			this.baseUrl = baseUrl;
+
+			PageFactory.initElements(driver, this);
+		}
+		
+		protected WebElement getProductPageElement(String productElement) {

```

Comment:
Move all of the locators in this class to the individual Page Objects. Reason is, these locators are properties of those pages. Your superclasses should only contain generalized functionality (which would apply to ALL Page Objects).